packages/commons/src/auth/jwt.ts

```diff
@@ -22,53 +39,62 @@ export const readAuthDataFromJwtToken = (jwtToken: string): AuthData => {
   }
 };
 
-const getKey =
-  (
-    clients: jwksClient.JwksClient[],
-    logger: Logger
-  ): ((header: JwtHeader, callback: SigningKeyCallback) => void) =>
-  (header, callback) => {
-    // eslint-disable-next-line functional/no-let
-    let responseReceived = 0;
-    for (const client of clients) {
-      client.getSigningKey(header.kid, function (err, key) {
-        responseReceived = responseReceived + 1;
-        if (err && responseReceived === clients.length) {
-          logger.error(`Error getting signing key: ${err}`);
-          return callback(err, undefined);
-        } else if (!err) {
-          return callback(null, key?.getPublicKey());
-        }
-      });
+const getKey = async (
+  clients: jwksClient.JwksClient[],
+  kid: string,
+  logger: Logger
+): Promise<Secret> => {
+  logger.info(`Getting signing key for kid ${kid}`);
+  for (const client of clients) {
+    try {
+      const signingKey = await client.getSigningKey(kid);
+      return signingKey.getPublicKey();
+    } catch (error) {
+      // Continue to the next client
+      logger.info(`Skip Jwks client`);
     }
-  };
+  }
+
+  logger.error(`Error getting signing key`);
+  return Promise.reject();
+};
 
-export const verifyJwtToken = (
+export const verifyJwtToken = async (
   jwtToken: string,
+  jwksClients: jwksClient.JwksClient[],
   logger: Logger
 ): Promise<boolean> => {
-  const config = JWTConfig.parse(process.env);
-  const clients = config.wellKnownUrls.map((url) =>
-    jwksClient({
-      jwksUri: url,
-    })
-  );
-  return new Promise((resolve, _reject) => {
-    jwt.verify(
-      jwtToken,
-      getKey(clients, logger),
-      {
-        audience: config.acceptedAudiences,
-      },
-      function (err, _decoded) {
-        if (err) {
-          logger.warn(`Token verification failed: ${err}`);
-          return resolve(false);
+  try {
+    const { acceptedAudiences } = JWTConfig.parse(process.env);
```

Comment:
See other comment about config

Reply:
This commit [1fbd47e](https://github.com/pagopa/interop-be-monorepo/pull/1002/commits/1fbd47ed1b96fc028d2dcb8b11e2f8433902466d) add configuration as parameter.